crates/bevy_ui/src/layout/mod.rs

```diff
@@ -276,95 +283,172 @@ pub fn ui_layout_system(
+        if node.calculated_size != rounded_size || node.unrounded_size != layout_size {
+            node.calculated_size = rounded_size;
+            node.unrounded_size = layout_size;
+        }
 
-            if let Some(outline) = maybe_outline {
-                // don't trigger change detection when only outlines are changed
-                let node = node.bypass_change_detection();
-                node.outline_width = outline
-                    .width
-                    .resolve(node.size().x, viewport_size)
-                    .unwrap_or(0.)
-                    .max(0.);
-
-                node.outline_offset = outline
-                    .offset
-                    .resolve(node.size().x, viewport_size)
-                    .unwrap_or(0.)
-                    .max(0.);
-            }
+        let viewport_size = root_size.unwrap_or(node.calculated_size);
 
-            if transform.translation.truncate() != rounded_location {
-                transform.translation = rounded_location.extend(0.);
-            }
+        if let Some(border_radius) = maybe_border_radius {
+            // We don't trigger change detection for changes to border radius
+            node.bypass_change_detection().border_radius =
+                border_radius.resolve(node.calculated_size, viewport_size);
+        }
+
+        if let Some(outline) = maybe_outline {
+            // don't trigger change detection when only outlines are changed
+            let node = node.bypass_change_detection();
+            node.outline_width = outline
+                .width
+                .resolve(node.size().x, viewport_size)
+                .unwrap_or(0.)
+                .max(0.);
+
+            node.outline_offset = outline
+                .offset
+                .resolve(node.size().x, viewport_size)
+                .unwrap_or(0.)
+                .max(0.);
+        }
 
-            if let Ok(children) = children_query.get(entity) {
-                for &child_uinode in children {
+        if transform.translation.truncate() != rounded_location {
+            transform.translation = rounded_location.extend(0.);
+        }
+
+        let scroll_position: Vec2 = maybe_scroll_position
+            .map(|scroll_pos| {
+                Vec2::new(
+                    if style.overflow.x == OverflowAxis::Scroll {
+                        scroll_pos.offset_x
+                    } else {
+                        0.0
+                    },
+                    if style.overflow.y == OverflowAxis::Scroll {
+                        scroll_pos.offset_y
+                    } else {
+                        0.0
+                    },
+                )
+            })
+            .unwrap_or_default();
+
+        let node_scrollable_bounds = rounded_size + approx_round_layout_coords(layout_location);
+
+        if let Ok(children) = children_query.get(entity) {
+            let mut children_bounding_box = children
```

Comment:
I believe this should already be computed by Taffy as the [`Layout::content_size`](https://docs.rs/taffy/latest/taffy/tree/struct.Layout.html#structfield.content_size) field which was created specifically for this use case (but hasn't been that extensively validated by real-world usage).

Is this computation different from that in some way?

Reply:
Nope, I just did not do my due diligence here! Updated to use it, which tamed most of the complexity.